Tagger/src/imageTag.java

```diff
@@ -61,10 +62,32 @@ private static void GetTagWhitelist() {
 					s = br.readLine();
 					while (s!=null) {
 						String newtag = s.trim().toLowerCase();
-						tag_whitelist.put(newtag,true);
-						System.out.println("Read in whitelisted tag: "+newtag);
+						String[] combtag;
+						List<String> listofsubs = new ArrayList<String>();

```

Comment:
This should be converted into a regular String. As every subtag should only have 1 main tag. List<String> is only required for lists, reduces complexity.